src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -98,7 +108,12 @@ void AdaptDispatch::_WriteToBuffer(const std::wstring_view string)
                 _DoLineFeed(textBuffer, true, true);
                 cursorPosition = cursor.GetPosition();
                 // We need to recalculate the width when moving to a new line.
-                state.columnLimit = textBuffer.GetLineWidth(cursorPosition.y);
+                lineWidth = textBuffer.GetLineWidth(cursorPosition.y);
+                if (cursorPosition.y >= topMargin && cursorPosition.y <= bottomMargin)
```

Comment:
As mentioned above, the left margin isn't relevant, and the reason we don't need to check the right margin is because we've just performed a carriage return/line feed (in the `_DoLineFeed` call), so we know we should definitely be inside the right margin.